pkgs/development/ocaml-modules/dose3/default.nix

```diff
@@ -0,0 +1,60 @@
+{ lib, buildDunePackage, fetchFromGitLab
+, camlzip, ocamlgraph, parmap, re, stdlib-shims
+, base64, bz2, ocaml_extlib, cudf
+, dpkg, git, ocaml, ounit, python39, python39Packages
+}:
+
+buildDunePackage rec {
+  pname = "dose3";
+  version = "6.1";
+
+  src = fetchFromGitLab {
+    owner = "irill";
+    repo = "dose3";
+    rev = version;
+    sha256 = "sha256-RFChY7VH2pVD4u5U1qng46h9aAv8I/3yXVaNmFDmKFI=";
+  };
+
+  minimalOCamlVersion = "4.03";
+  useDune2 = true;
+
+  buildInputs = [
+    base64
+    cudf
+    parmap
+    re
+  ];
+  propagatedBuildInputs = [
+    bz2
+    camlzip
+    ocaml_extlib
+    ocamlgraph
+    stdlib-shims
+  ];
```

Comment:
How did you decide to propagate these and not the other ones?

Reply:
For instance, `parmap` must NOT be propagated, as it is only used for building the `dose-challenged` binary.

Reply:
Initially I was propagating everything. After you hinted to not do this, I started propagating nothing unless something downstream would error. The packages I have moved to propagated are the ones that showed up on downstream errors so far. I still haven't mastered the `ocamlfind` method to solve this problem. I honestly don't know in advance what to propagate or not.